### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by Headieh**

### CODE REVIEW

**Comment by Headieh** [comment]
Not sure if I updated my forked copy correctly with this commit (047e418) in this pull request

File: students/headieh_godwin/lesson04/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+import inquirer
+import os
+import time
+
+donor1 = {
+    "name"        : "Karen",
+    "donations"   :  (20,20,100)
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:8]: Missing whitespace after commas here and below

File: students/headieh_godwin/lesson04/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+import inquirer
+import os
+import time
+
+donor1 = {
+    "name"        : "Karen",
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:7]: No whitespaces before colons

File: students/headieh_godwin/lesson04/mailroom.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+2 - Create a Report.
+3 - Send letters to all donors.
+4 - Quit
+""")
+    return(answers)
+
+def email(x, y, z):
+        
+    if z == 0: # case for adding donor
+        print ( f"""
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:78]: No whitespaces before or after parentheses

File: students/headieh_godwin/lesson04/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+Dear {x},
+
+Thank you for your very kind donation of ${y:.2f}
+     
+It will be put to very good use.
+
+                       Sincerely,
+                          -The Team""") 
+    if z == 1: #case for sending emails to all who have donated one time
+        return ( f"""
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:88]: Here as well

File: students/headieh_godwin/lesson04/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+            filters = [response1 in names for names in donors]
+            filters = [i for i, x in enumerate(filters) if x][0]
+            wo_d = donors[filters][1]
+            response2 = float(response2)
+            if(type(wo_d) is tuple):
+                w_d = (wo_d) + (response2,)
+            else:
+                w_d = (wo_d,) + (response2,)
+            donors[filters][1] = w_d
+            email(response1, response2, z=0)
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:150]: Isn't adding to existing donations

File: students/headieh_godwin/lesson04/mailroom.py [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            n = len(v['donations'])
+        else:
+            tot = v['donations']
+            n = 1
+        avg = round(tot/n,2)
+        raw.append({'name'      : v['name'],
+                    'total'     : tot,
+                    'number'    : n,
+                    'average'   : avg})
+    sort_data = (sorted(raw, key = lambda i: i['total']))
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson04/mailroom.py:195]: Donors should be sorted in descending order

**Comment by charlesrobison** [comment]
@Headieh - just tried to run trigrams and got an error.  Is this ready for review?  If so, please provide the file your running it with if it's working for you - thanks!

**Comment by Headieh** [comment]
I was running it with the sherlock.txt linked on here
 https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/kata_fourteen.html

**Comment by charlesrobison** [comment]
Ah, got it.  Works now.  Nice job!

@natasha-aleksandrova  - this PR is good to go

**Comment by Headieh** [comment]
@natasha-aleksandrova @charlesrobison I'm still not sure I updated my fork correctly - its the 047e418 commit in this PR though

**Comment by natasha-aleksandrova** [comment]
@Headieh your fork looks good 👍

**Pull request merged**